### DISCUSSION TITLE
I cannot define a multi-select options field for a column on a postgreSQL schema

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#11069] [Help]
I have a schema that has a table with a column called `languages` 
It has a data type in postgreSQL of `language[]` where `language` is a type defined with labels like `en, fr, de, es, it, pl, pt, ro, ru, tr, uk`
postgreSQL stores the data as a string in the field - like this `{en,fr}` for example.

I want to edit this field in a Budibase application.   The only component I can find that facilitates this sort of thing is a `multi-select picker`. However if I add such a component and associate it with the field from the data-provider, it just says "The Field setting is the wrong data type for this component".  It doesn't say what the valid types are (not even in the docs).

The column has a data type of `text` so what should it be?   And how is BB going to handle the fact that postgres likes this data stored in a comma delimited list surrounded by braces?

I would hazard a guess that the only field type allowed is a type of "multi-select" but you cannot change a column on an imported schema to `mult-select`. Is there a workaround?

**Comment by kpturner** [comment]
Whilst waiting for a definitive answer, I embarked on an experiment.

In the maintenance screen I added a `multi-select picker` field and called in `language_worker`.  
I modified the `Default value` to be a Javascript function: `return $("Form.Fields.languages").replace('{','').replace('}','')` and that works OK.
To that I added some options.
In this fields `On change` event I defined an `Update Field Value` action with a view to populating the actual `languages` field for me.

I did this with a JavaScript function like so: `return '{'+$("Form.Fields.languages_worker")+'}'`
Whilst this does populate the `languages` field OK, it is always one selection behind.  The value of `$("Form.Fields.languages_worker")` in the On Change event is the value BEFORE the option was selected, not after.

So when I select, for arguments sake, `fr` this javascript function returns `{}`
If I also select `en` the Javascript function returns `[fr}`
If I also select `uk` the Javascript function returns `{fr,en}`

Is this a bug?   The same thing happens if I return `{{ Form.Fields.languages_worker }}` as a text value rather than use JavaScript.

I have worked around this by making the change in the "save" event instead of the on-change event of the field.

**Reply by melohagan** [reply]
Hey @kpturner Part of the problem is that we don't have JSON support in Postgres as yet: https://github.com/Budibase/budibase/issues/3382

**Reply by JCV67** [reply]
In the Budibase fetched table, on the options type column, you need to create the same options you defined on Postgres.

**Reply by kpturner** [reply]
Not sure how that helps? 
It just saves them as a comma delimited string plus you can't make an imported column a multi-select as far as I can see?

**Comment by m7kael** [comment]
I have also have issues with multi-select, I have created the columns in bb and it has work well until I had to refetch the tables. Now I cannot define the type and the form fields all have stopped working. Anyone have any suggestions?

<img width="303" alt="Screenshot 2023-07-11 at 21 52 47" src="https://github.com/Budibase/budibase/assets/112936118/66bb3679-9f63-418d-9224-be5d43a244c5">
